frameworks/hdfs/tests/test_tls.py

```diff
@@ -106,6 +110,31 @@ def fn():
     assert fn()
 
 
+@pytest.mark.tls
+@pytest.mark.sanity
+@pytest.mark.recovery
+def test_tls_recovery(hdfs_service_tls, service_account):
+    try:
+        pod_name = "name-0"
+        inital_task_id = sdk_tasks.get_task_ids(config.SERVICE_NAME, pod_name)
+
+        cmd_list = [
+            "pod", "replace", pod_name,
+        ]
+        sdk_cmd.svc_cli(config.PACKAGE_NAME, config.SERVICE_NAME,
+                        " ".join(cmd_list))
+
+        recovery_timeout_s = 25 * 60
+        sdk_plan.wait_for_kicked_off_recovery(config.SERVICE_NAME, recovery_timeout_s)
+        sdk_plan.wait_for_completed_recovery(config.SERVICE_NAME, recovery_timeout_s)
+
+        sdk_tasks.check_tasks_updated(config.SERVICE_NAME, pod_name, inital_task_id)
+
+        # TODO: Add checks for non-updated tasks
+    finally:
+        hdfs_service_tls(service_account)
```

Comment:
I don't get this. Why do it in the finally block?

Reply:
For the record, I have removed this.